x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java

```diff
@@ -1309,7 +1309,12 @@ public enum Cap {
         /**
          * Support correct counting of skipped shards.
          */
-        CORRECT_SKIPPED_SHARDS_COUNT;
+        CORRECT_SKIPPED_SHARDS_COUNT,
+
+        /**
+         * Byte elements dense vector field type support.
+         */
+        DENSE_VECTOR_FIELD_TYPE_BYTE_ELEMENTS(EsqlCorePlugin.DENSE_VECTOR_FEATURE_FLAG);
```

Comment:
this make sense to me for now - when we get to take these features out of snapshot, maybe we can create single capability to encompass all the features, from dense_vector support to the knn function and brute force functions and we can remove the individual ones we have created for each step.